drift/lib/src/runtime/query_builder/expressions/aggregate.dart

```diff
@@ -112,6 +112,49 @@ extension ArithmeticAggregates<DT extends num> on Expression<DT?> {
       _AggregateExpression('TOTAL', [this], filter: filter);
 }
 
+/// Provides aggregate functions that are available for BigInt expressions.
+extension BigIntAggregates<DT extends BigInt> on Expression<DT?> {
```

Comment:
I'm not entirely sure whether we should expose these numeric expressions for `BigInt` as well since it's the same thing as a normal integer in SQL. I think the main convenience is  that drift automatically reports those values as `BigInt`s without any loss of precision along the way, but that's it.

So if users want to do arithmetic on BigInts, IMO they can do `(foo.dartCast<int>() + bar.dartCast<int>()).dartCast<BigInt>()`. Sure, having that in drift is way more convenient, but

- using `BigInt`s is not something that most users will want to deal with in the end (it's only useful for columns known to store values larger than 2⁵², so typically large IDs on which one doesn't do arithmetic a lot)
- we realistically need to maintain those expressions separately, with integration tests and everything

I see how this isn't ideal for few users, but I think we can make our lives easier here by forcing users to go the indirect way through `dartCast`.

Reply:
I do not agree. Casting to int is against the sense of BigInt. I understand using BigInt rather edge case and affect only few users. But skipping BigInt aggregations and arithmetics will make BigInt support partial and users won't understand why operations do not work on some columns. I think these two class doesn't make things more complicated compared to the whole project.

Reply:
> Casting to int is against the sense of BigInt

Is it though? In SQL, a `Expression<BigInt>` and `Expression<int>` is the same thing. The only difference is how that value is mapped from and to Dart. So casting intermediate values to `Expression<int>` will not change the query in any way.

Perhaps we can keep the new API but use casts around the existing methods internally (it will generate the same SQL), I'll think about it some more. If it turns out to be easy to support, I agree that it's better to have these methods. I'm mostly worrying about duplicate code that's easy to forget about when making changes to the original APIs for `int`.

Reply:
Yeah. Sorry. I forget it is just a DSL for SQLite. And this internal cast would be nice. The BigInt aggreg is casted to regular int.

Reply:
I think it should be fine for postgres too since the generated SQL is the same and since those aggregates also work on `bigint`s in postgres?